Rory cluster

rodwyer100 opened the pull request:
The most recent version of SvtRawDataAnaProcessor I was using. Furthermore the new Cluster processors

rodwyer100 committed:
Add files via upload

rodwyer100 committed:
Add files via upload

rodwyer100 committed:
Add files via upload

sarahgaiser commented on processors/config/roryClusterAna.py:
Maybe we can re-name the files somehow? Once Rory graduates, nobody will understand why the files are called rory... It should be more descriptive of what the file/program is doing.

rodwyer100 replied:
Done

sarahgaiser commented on processors/config/roryClusterAna.py:
```diff
@@ -0,0 +1,56 @@
+base.parser.add_argument("-MC", "--MC", type=int, dest="isMC",
+                  help="IS IT MC", metavar="module",default=0)
+base.parser.add_argument("-doT", "--doT", type=int, dest="doTrack",
+                  help="DO WE DO TRACK STUFF", metavar="doTrack",default=0)
```
Let's be a bit less casual here. This could be "using tracking information" or sth of that sort (idk what exactly your program is doing so please adjust this)

rodwyer100 replied:
done

sarahgaiser commented on processors/config/roryClusterCompareAna.py:
```diff
@@ -0,0 +1,56 @@
+                  help="Module Under Investigation", metavar="module",default=-1)
+base.parser.add_argument("-MC", "--MC", type=int, dest="isMC",
+                  help="IS IT MC", metavar="module",default=0)
+base.parser.add_argument("-doT", "--doT", type=int, dest="doTrack",
```
same as above

rodwyer100 replied:
done

sarahgaiser commented on processors/src/RoryClusterAnaProcessor.cxx:
```diff
@@ -0,0 +1,677 @@
+    return BigCount;
+}
+
+void RoryClusterAnaProcessor::Plot1(){
```
Function names should always be describing what the function is doing. plot1 doesn't tell me much about the content or purpose of plot1 vs plot2 vs plot3 etc

rodwyer100 replied:
Done

sarahgaiser commented on processors/src/RoryClusterAnaProcessor.cxx:
```diff
@@ -0,0 +1,677 @@
+}
+
+void RoryClusterAnaProcessor::Plot1(){
+    std::cout<<"I AM IN THE FINAL STEP"<<std::endl; 
```
also, print statements like this are nice for development but should not be in the final version. If you want to keep this as a debug statement or just for keeping track of the progress of your program, please adjust it to sth along the lines of "RoryClusterAnaProcessor::Plot1:: statement here"

rodwyer100 replied:
Done

sarahgaiser commented on processors/src/RoryClusterAnaProcessor.cxx:
```diff
@@ -0,0 +1,677 @@
+    return;
+}
+
+void RoryClusterAnaProcessor::Plot2(){
```
change name

sarahgaiser commented on processors/src/RoryClusterAnaProcessor.cxx:
```diff
@@ -0,0 +1,677 @@
+    return;
+}
+
+void RoryClusterAnaProcessor::Plot3(){
```
change name

rodwyer100 replied:
Done

sarahgaiser commented on processors/src/RoryClusterAnaProcessor.cxx:
```diff
@@ -0,0 +1,677 @@
+    return;
+}
+
+void RoryClusterAnaProcessor::Plot4(){
```
you get the point

sarahgaiser commented on processors/src/RoryClusterAnaProcessor.cxx:
```diff
@@ -0,0 +1,677 @@
+    return;
+}
+
+void RoryClusterAnaProcessor::TrackPlot1(){
```
this name is almost good. maybe remove the 1 because there is only one track plot as far as I can tell?

rodwyer100 replied:
done

sarahgaiser commented on processors/src/RoryClusterCompareAnaProcessor.cxx:
```diff
@@ -0,0 +1,717 @@
+RoryClusterCompareAnaProcessor::RoryClusterCompareAnaProcessor(const std::string& name, Process& process) : Processor(name,process){
+    mmapper_ = new ModuleMapper(2021);
+}
+//TODO CHECK THIS DESTRUCTOR
```
is this still an active todo?

rodwyer100 replied:
Nope

sarahgaiser commented on processors/src/RoryClusterCompareAnaProcessor.cxx:
```diff
@@ -0,0 +1,717 @@
+    return BigCount;
+}
+
+void RoryClusterCompareAnaProcessor::Plot1(){
```
change the name of this

rodwyer100 replied:
Done

sarahgaiser commented on processors/src/RoryClusterCompareAnaProcessor.cxx:
```diff
@@ -0,0 +1,717 @@
+    return;
+}
+
+void RoryClusterCompareAnaProcessor::Plot2(){
```
change name

rodwyer100 replied:
Done

sarahgaiser commented on processors/src/RoryClusterCompareAnaProcessor.cxx:
```diff
@@ -0,0 +1,717 @@
+    return;
+}
+
+void RoryClusterCompareAnaProcessor::TrackPlot1(){
```
what's the difference between trackplot1 and trackplot2? I should be able to tell this from the function name

rodwyer100 replied:
Done

sarahgaiser commented on processors/src/SvtRawDataAnaProcessor.cxx:
```diff
@@ -210,10 +210,12 @@ void SvtRawDataAnaProcessor::initialize(TTree* tree) {
     //histos = new RawSvtHitHistos(anaName_.c_str(), mmapper_);
     //histos->loadHistoConfig(histCfgFilename_);
     //histos->DefineHistos();
+    //std::cout<<"hello4"<<std::endl;
```
remove this

rodwyer100 replied:
Done

sarahgaiser commented on processors/src/SvtRawDataAnaProcessor.cxx:
```diff
@@ -251,46 +261,156 @@ void SvtRawDataAnaProcessor::initialize(TTree* tree) {
     bool doClMatch = true;
     
-    
+    int STR = -10000;
```
usually, variables are lower case

sarahgaiser replied:
also, is this supposed to be a strip ID or sth? in that case, please put that in the name

rodwyer100 replied:
DONE

sarahgaiser commented on processors/src/SvtRawDataAnaProcessor.cxx:
```diff
@@ -251,46 +261,156 @@ void SvtRawDataAnaProcessor::initialize(TTree* tree) {
     //std::cout<<"Trigger Time: "<<vtpBank_->singletrigs.at(0).T<<std::endl;
+    
+    
+    //std::cout<<"I got here 9B"<<std::endl;
```
remove all old debug statements, unless they might be needed in the future

rodwyer100 replied:
Done

sarahgaiser commented on processors/src/SvtRawDataAnaProcessor.cxx:
```diff
@@ -372,31 +500,53 @@ bool SvtRawDataAnaProcessor::process(IEvent* ievent) {
-     *
-     */
+/* 
+int SvtRawDataAnaProcessor::hitEff(IEvent* ievent, int L){
```
remove unused code

rodwyer100 replied:
Done

sarahgaiser commented:
I will give this a second round of review once you are done cleaning up your code. Please remove all old/unused print statements and functions that are commented out and no longer needed. It would be great if you could also document your code. That means including a short statement about the purpose of a function or class, input and output of the function and general nice-to-know things about it. For classes, you usually also add short comments about the class attributes.
If you want to have a reference for including comments, you can refer to [this](https://google.github.io/styleguide/cppguide.html#Comments).

rodwyer100 committed:
Update and rename RoryClusterAnaProcessor.cxx to ClusterAnaProcessor.cxx

rodwyer100 committed:
Update and rename RoryClusterAnaProcessor.h to ClusterAnaProcessor.h

rodwyer100 committed:
Update RoryClusterCompareAnaProcessor.cxx

rodwyer100 committed:
Rename RoryClusterCompareAnaProcessor.cxx to ClusterCompareAnaProcessor.cxx

rodwyer100 committed:
Update and rename RoryClusterCompareAnaProcessor.h to ClusterCompareAnaProcessor.h

rodwyer100 committed:
Update and rename roryClusterAna.py to ClusterAna.py

rodwyer100 committed:
Rename ClusterAna.py to clusterAna.py

rodwyer100 committed:
Update and rename roryClusterCompareAna.py to clusterCompareAna.py

rodwyer100 committed:
Update ClusterAnaProcessor.cxx

rodwyer100 committed:
Update ClusterAnaProcessor.cxx

rodwyer100 committed:
Update ClusterAnaProcessor.h

rodwyer100 committed:
Update ClusterCompareAnaProcessor.cxx

rodwyer100 committed:
Update ClusterCompareAnaProcessor.h

rodwyer100 committed:
Update SvtRawDataAnaProcessor.cxx

rodwyer100 committed:
Update SvtRawDataAnaProcessor.cxx

rodwyer100 committed:
Update ClusterAnaProcessor.cxx

rodwyer100 committed:
Update ClusterCompareAnaProcessor.cxx

rodwyer100 committed:
Update ClusterAnaProcessor.cxx

rodwyer100 committed:
Update ClusterCompareAnaProcessor.cxx

rodwyer100 committed:
Add files via upload

rodwyer100 committed:
Update AddIdentity.C

rodwyer100 committed:
Update clusterCompareAna.py

rodwyer100 committed:
Update clusterAna.py

sarahgaiser commented on processors/config/clusterAna.py:
```diff
@@ -0,0 +1,56 @@
+
+p.sequence = sequence
+
+print("THE INPUT FILE IS "+str(root_file))
```
maybe "processors::clusterAna: The input file is ..." is better? @cbravo135 is there a standard format for these messages?

rodwyer100 replied:
I have changed it in anticipation of this being the right answer.

sarahgaiser commented on processors/config/clusterCompareAna.py:
```diff
@@ -0,0 +1,56 @@
+
+p.sequence = sequence
+
+print("THE INPUT FILE IS "+str(root_file))
```
see above

rodwyer100 replied:
Changed that :)

rodwyer100 commented:
I believe I have replied to your comments :). Also I added a root macro to the scripts collection to facilitate usage of the ClusterCompareAnaProcessor.cxx

sarahgaiser commented on processors/src/ClusterAnaProcessor.cxx:
```diff
@@ -0,0 +1,672 @@
+
+/**
+ *
+ *THIS METHOD FILLS SEVERAL HISTOGRAMS FOR EACH EVENT DIRECTED AT EVALUATING CLUSTERING PERFORMANCE. FOR OUR TRACKING
```
Great, thank you for adding this. Usually, the detailed description of the function goes with the declaration of the function in the header file. Idk why it's done like that but let's stick with it

rodwyer100 replied:
Got it :)

sarahgaiser commented on processors/src/ClusterAnaProcessor.cxx:
```diff
@@ -0,0 +1,672 @@
+    }
+    for(int i = 0; i < Clusters_->size(); i++){ 
+        TrackerHit * clu = Clusters_->at(i);
+        Int_t LAYER = -1;
```
lower case

rodwyer100 replied:
Done

sarahgaiser commented on processors/src/ClusterAnaProcessor.cxx:
```diff
@@ -0,0 +1,672 @@
+        }
+        bool general = ((layer_==-1)||(module_==-1));
+        if(((LAYER==layer_)&&(MODULE==module_))||(general)){
+            //NOW IS THE PART WHERE I FILL THE CLUSTER DISTANCE HISTOGRAM
```
could you write this in not ALL CAPS

rodwyer100 replied:
Done

sarahgaiser commented on processors/src/ClusterAnaProcessor.cxx:
```diff
@@ -0,0 +1,672 @@
+                ClusDistances_->Fill(Dist);
+                if(NTD){ClusDistancesNTD_->Fill(Dist);}
+            }
+            //std::cout<<"HELLO"<<std::endl;
```
remove

rodwyer100 replied:
done

sarahgaiser commented on processors/src/ClusterCompareAnaProcessor.cxx:
```diff
@@ -0,0 +1,728 @@
+ClusterCompareAnaProcessor::ClusterCompareAnaProcessor(const std::string& name, Process& process) : Processor(name,process){
+    mmapper_ = new ModuleMapper(2021);
+}
+//TODO CHECK THIS DESTRUCTOR
```
is this still an active TODO?

rodwyer100 replied:
done

sarahgaiser commented on processors/src/SvtRawDataAnaProcessor.cxx:
```diff
@@ -255,54 +215,104 @@ bool SvtRawDataAnaProcessor::process(IEvent* ievent) {
+                                //THIS CONDITIONS ON IT BEING IN CLUSTERS
+                                
+                                bool InCluster = false;
+                                int LAY = 0;//THE PURPOSE OF LAY IS TO COUNT THE NUMBER OF HITS PER LAYER
```
lower case

rodwyer100 replied:
done

sarahgaiser reviewed (changes requested):
Thanks for addressing my earlier comments! I found a few more minor formal things that could be improved. Usually, the longer comment/documentation of a function goes in the header file (if I'm not mistaken) so maybe we could also move it there. Additionally, it would be great if you could rename some variables to lower case names. And instead of using all caps for comments, please just use normal capitalization.
Other than that, there are a few formatting things that I don't want to highlight here but that we can talk about in person. (Mostly about leaving spaces around = and other operators but we can fix that automatically somehow.)

sarahgaiser commented:
I can do a more functional review early next week but I assume you and Cam already tested that in the last few weeks so idk if there is much use in me doing that. I will def look over the code at least one more time though and look out for typos, bugs, and other things that I haven't looked for yet.

rodwyer100 committed:
Update ClusterAnaProcessor.cxx

rodwyer100 committed:
Update ClusterCompareAnaProcessor.cxx

rodwyer100 committed:
Update ClusterCompareAnaProcessor.cxx

rodwyer100 committed:
Update SvtRawDataAnaProcessor.cxx

rodwyer100 committed:
Update ClusterCompareAnaProcessor.cxx

rodwyer100 committed:
Update ClusterAnaProcessor.h

rodwyer100 committed:
Update ClusterAnaProcessor.cxx

rodwyer100 committed:
Update ClusterAnaProcessor.h

rodwyer100 committed:
Update ClusterCompareAnaProcessor.h

rodwyer100 committed:
Update SvtRawDataAnaProcessor.h

rodwyer100 committed:
Update SvtRawDataAnaProcessor.cxx

rodwyer100 committed:
Update clusterCompareAna.py

rodwyer100 committed:
Update clusterAna.py

rodwyer100 commented:
Made a second round of corrections as per your suggestions. I have changed the config printout for input files to your convention in anticipaton of it being the 'official' way to do things :)! I believe I have addressed everything else as well.

sarahgaiser commented on processors/src/SvtRawDataAnaProcessor.cxx:
```diff
@@ -234,75 +170,114 @@ void SvtRawDataAnaProcessor::initialize(TTree* tree) {
+        int getNum = thisHit->getFitN();
         if(doClMatch){
             bool Continue = true;
             for(int i = 0; i<Part_->size();i++){
```
you have two for loops running over i here

rodwyer100 replied:
Fixed that. I will need to investigate if it affected any of my studies. I would need to know what happens when you have an unsigned int and a signed int with the same name, when you increment one in the encased loop whether it increments both or what it does. Good catch, wish I got it earlier sigh,

rodwyer100 replied:
Looks like I got lucky. It seems to refer to the variable with the most recent usage in the interior loop, so I won't need to redo a couple studies.

sarahgaiser commented on processors/src/ClusterCompareAnaProcessor.cxx:
```diff
@@ -0,0 +1,683 @@
+            if(ident_<1.5){TrackMomentumTAllTime1_->Fill(track->getP());}else{TrackMomentumTAllTime2_->Fill(track->getP());}
+        
+            if(track->getTrackTime()*track->getTrackTime()<16.0){
+                if(ident_<1.5){TrackMomentumTInTime1_->Fill(track->getP());}else{TrackMomentumTInTime2_->Fill(track->getP());}
```
did you mean to use getPt() here? because it says that you are getting the transverse momentum

rodwyer100 replied:
Yep thanks!

sarahgaiser commented on processors/src/ClusterCompareAnaProcessor.cxx:
```diff
@@ -0,0 +1,683 @@
+    for(int i = 0;i<24576;i++){
+        Deads_[i]=0.0;
+    }
+    std::string FILENAME="/sdf/group/hps/users/rodwyer1/run/cluster_study/badchannels2021.dat";
```
Maybe we can put the file with bad channels somewhere central so that we don't have to include this very specific path here?

rodwyer100 replied:
I put is in processors/dat and included it as a config argument of the python file. I think thats the most canonical way of doing things. I will change the processor names soon as per Cam's suggestion :)

sarahgaiser committed:
formatting

sarahgaiser reviewed (commented):
While formatting some things, I had these additional questions/comments. Could you address those in a comment rather than fixing them directly? I would just like to push my changes before we push sth on the branch from another end and applying my changes gets super complicated.

rodwyer100 committed:
Update SvtRawDataAnaProcessor.cxx

rodwyer100 committed:
Update ClusterCompareAnaProcessor.cxx

rodwyer100 committed:
Add files via upload

rodwyer100 committed:
Update clusterAna.py

rodwyer100 committed:
Update clusterCompareAna.py

rodwyer100 committed:
Update ClusterCompareAnaProcessor.cxx

rodwyer100 committed:
Update ClusterCompareAnaProcessor.h

rodwyer100 committed:
Update ClusterAnaProcessor.h

rodwyer100 committed:
Update ClusterAnaProcessor.cxx

rodwyer100 committed:
Update and rename ClusterCompareAnaProcessor.h to TrackHitCompareAnaProcessor.h

rodwyer100 committed:
Update TrackHitCompareAnaProcessor.h

rodwyer100 committed:
Update TrackHitCompareAnaProcessor.h

rodwyer100 committed:
Update and rename ClusterCompareAnaProcessor.cxx to TrackHitCompareAnaProcessor.cxx

rodwyer100 committed:
Update and rename ClusterAnaProcessor.h to SVTClusterAnaProcessor.h

rodwyer100 committed:
Update and rename ClusterAnaProcessor.cxx to SVTClusterAnaProcessor.cxx

rodwyer100 committed:
Update and rename clusterCompareAna.py to trackHitCompareAna.py

rodwyer100 committed:
Update and rename clusterAna.py to trackHitClusterAna.py

rodwyer100 commented:
Okay I changed the names to names Cam suggested, though I had to get a little creative as he already created a processor called 'TrackHitAnaProcessor" which I didn't know about. One is now called "TrackHitCompareAnaProcessor" while the other is "SVTClusterAnaProcessor" as it would have had the already used name; both of these were approved names for the object by Cam. I addressed the double usage of the for loop indexer and verified that it did not affect the operation of the processor as expected; I changed it anyways as it is confusing and potentially dangerous. I also addressed the hardcoded location of bad channels by putting it into a dat collection and I believe thats everything so far :)!

sarahgaiser committed:
small change

sarahgaiser commented:
I have updated the formatting of your code. Can you check if everything builds and if the output is as expected? Once you got that done, I'll approve.

rodwyer100 committed:
Add files via upload

rodwyer100 committed:
Add files via upload

rodwyer100 committed:
Add files via upload

rodwyer100 committed:
Add files via upload

rodwyer100 commented:
I got it so it compiles and does what it should do. Had to add some changes from RawSvtHitHistos.cxx in analysis to do it, but they ought to be minor.

sarahgaiser committed:
last formatting touches

sarahgaiser committed:
Delete .vscode/settings.json

sarahgaiser reviewed (approved):
Looks good to me now!

cbravo135 commented on scripts/AddIdentity.C:
```diff
@@ -0,0 +1,18 @@
+/**
```
Remove this file, such an identifier already exists in the event header.

rodwyer100 replied:
This identifier is used to identify separate reconstruction techniques. Is that in the event header?

cbravo135 replied:
you could easily do what you want with the existing header, yes

cbravo135 replied:
You just have to fill it correctly

cbravo135 reviewed (changes requested):
Remove scripts/AddIdentity.C

cbravo135 commented on processors/dat/badchannels2021.dat:
This file should be somewhere in analysis, not processors.

rodwyer100 replied:
Done, put in analysis/data

cbravo135 commented on processors/config/trackHitCompareAna.py:
```diff
@@ -0,0 +1,57 @@
+###############################
+#          Processors         #
+###############################
+cclua = HpstrConf.Processor('cclua', 'TCompareAnaProcessor')
```
I am pretty sure TCompareAnaProcessor is a typo.

rodwyer100 replied:
fixed

rodwyer100 committed:
Update trackHitCompareAna.py

rodwyer100 committed:
Add files via upload

rodwyer100 committed:
Add files via upload

rodwyer100 committed:
Delete processors/dat/badchannels2021.dat

rodwyer100 committed:
Update trackHitCompareAna.py

rodwyer100 committed:
Update trackHitClusterAna.py
